pkg/alerter/alerter.go

```diff
@@ -0,0 +1,79 @@
+package alerter
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/sirupsen/logrus"
+)
+
+type Alerter struct {
+	sync.Mutex
+	Alerts   Alerts
+	AlertsCh chan Alerts
+}
+
+func NewAlerter() *Alerter {
+	return &Alerter{
+		Alerts:   make(Alerts),
+		AlertsCh: make(chan Alerts),
+	}
+}
+
+func (a *Alerter) Start() {
+	go func() {
+		defer close(a.AlertsCh)
+		for {
+			alert := <-a.AlertsCh
+			a.mergeAlerts(alert)
+		}
+	}()
+}
+
+func (a *Alerter) Alert(key string, alert Alert) {
+	a.AlertsCh <- Alerts{
+		key: []Alert{alert},
+	}
+}
+
+func (a *Alerter) mergeAlerts(alert Alerts) {
+	a.Lock()
+	defer a.Unlock()
+	for k, v := range alert {
+		if val, ok := a.Alerts[k]; ok {
+			a.Alerts[k] = append(val, v...)
+		} else {
+			a.Alerts[k] = v
+		}
+	}
+}
+
+func (a *Alerter) IgnoreResources(resources []resource.Resource) []resource.Resource {
+	results := make([]resource.Resource, 0, len(resources))
+
+	for _, res := range resources {
+		alert, alertExists := a.Alerts[fmt.Sprintf("%s.%s", res.TerraformType(), res.TerraformId())]
+		wildcardAlert, wildcardAlertExists := a.Alerts[res.TerraformType()]
+		if (!alertExists && !wildcardAlertExists) ||
+			(!a.shouldBeIgnored(alert) && !a.shouldBeIgnored(wildcardAlert)) {
+			results = append(results, res)
+		} else {
+			logrus.WithFields(logrus.Fields{
+				"type": res.TerraformType(),
+				"id":   res.TerraformId(),
+			}).Debug("Found resource to ignore because of an alert")
+		}
+	}
+
+	return results
+}
+
+func (a *Alerter) shouldBeIgnored(alert []Alert) bool {
```

Comment:
It could be a method of `Alerts` struct directly

```go
func (a *Alerts) shouldBeIgnored(resource resource.Resource) bool {
  for _, v := range a {
      // Check if the resource match and if it should be ignored
  }
} 
```

Reply:
We could do a method of Alerts but I don't get it why we may need a loop here.

Since it's a map of string []Alert, we could simply check if both the single `key.id` and the wildcard `key` exist, and retrieve if either one those two has an Alert inside it that has the bool ShouldIgnoreResource.

cf. what i did on the `IgnoreResources` or the `Run` of the drigtignore.

Reply:
What about something like this ?

![image](https://user-images.githubusercontent.com/6154987/104195637-2bc74400-5423-11eb-8870-a90b2301957c.png)

Reply:
It would mean that if by any chance we have this kind of setup for alerts, it won't be ignored:

```
"alerts": {
  "aws_resource": [
    { "Message": "something", "ShouldIgnoreResource": true },
  ],
  "aws_resource.foo": [
    { "Message": "updated.field is a computed field" },
  ]
}
```

Reply:
Right nice catch, we need to invert the two alerts assignations to ensure correct precedence 

![image](https://user-images.githubusercontent.com/6154987/104205791-19530780-542f-11eb-81a8-0cc0ce7bbd29.png)